src/wazuh_testing/modules/aws/data_generator.py

```
@@ -1167,13 +1192,14 @@ def get_data_sample(self):
 }
 
 
-def get_data_generator(bucket_type, bucket_name, creation_date) -> DataGenerator:
+def get_data_generator(bucket_type, bucket_name, creation_date, region) -> DataGenerator:
```

Now that the region is an input, could we validate that it is valid and fail early? Something like

```py
from boto3.session import Session
...

session = Session()
service_regions = session.get_available_regions('<service>')
if region not in service_regions:
    raise Exception(f"Invalid region, available regions are: {', '.join(service_regions)}")
```

---

Done